src/Microcosm.js

```diff
@@ -71,31 +49,34 @@ Microcosm.prototype = {
    * a new state. This is the state exposed to the outside world.
    */
   rollforward() {
-    let next = this.base
+    this.state = this.history.pathway().reduce(dispatch.bind(this, this.stores), this.base)
+    this.emit(this.state)
 
-    for (var i = 0; i < this.transactions.length; i++) {
-       if (this.transactions[i].active) {
-         next = this.dispatch(next, this.transactions[i])
-       }
-     }
+    return this
+  },
 
-    if (next !== this.state) {
-      this.state = next;
-      this.emit(next)
-    }
+  clean(transaction, next) {
+    if (!transaction.complete) return
 
-    return this
+    this.base = dispatch(this.stores, this.base, transaction)
+    this.history.remove(transaction)
+
+    next()
   },
```

Comment:
This is the default merger strategy. It means: "progress through the current transaction pathway, merging transactions that are complete. Stop when you it an incomplete transaction"